platform-sdk/docs/proposals/consensus-layer/Consensus-Layer.md

```diff
@@ -213,57 +233,90 @@ passed to Event Intake as part of the event metadata.
+persisted before being gossiped. While it may not be necessary for all code paths to have durable pre-consensus events
+before they can handle them, to simplify the understanding of the system, we simply make all events durable before
+distributing them. This leads to a nice, clean, simple understanding that, during reply, the entire system will
+behave predictably.
 
 #### Roster Changes
 
 Since Event Intake must validate events, and since event validation requires knowing the roster (to verify the event
 source is in the roster, and the gossip source is in the roster, and the signature of the event creator is correct),
-Event Intake must know about changes to the roster. As with Gossip, it is imperative that Event Intake maintain a
-history of rosters, so it can support peers that are farther behind in consensus than it is.
-
-### Pre-Consensus Recording Module
-
-The Pre-Consensus Recording module is responsible for recording ordered, valid events which have not yet come to
-consensus. This is critical for minimizing data loss if catastrophic network failure occurs. This module received
-simplification by finding methods that durably persist events fast enough that it can be used inline. That is, each
-event is persisted before it is gossiped, and before it is added to the Hashgraph. The delay introduced by this module
-**must** be minimized to keep gossip latency minimal and increase the event/sec throughput of the system as a whole.
-
-This system is further simplified by defining its storage as a cyclic buffer. For example, it may have a large file on
-disk, and maintain a "head" and "tail" pointer into that file. Then, as new events need to be persisted, it simply
-writes them in at the "tail" position and moves forward. As it reaches the end of the file, it loops back to the start.
-In this way, it is able to have a fixed buffer, large enough for disaster recovery, but without requiring any feedback
-loops from Execution to indicate when data may be purged. Typically, this would be configured to be some value several
-multiples in size larger than the state saving timeframe and maximum event/sec rate, so as to provide a solid guarantee
-of data availability.
+Event Intake must know about changes to the roster. It is imperative that Event Intake maintain a history of rosters,
+so it can support peers that are farther behind in consensus than it is.
+
+### Hashgraph Module
+
+The Hashgraph module orders events into rounds, and assigns timestamps to events. It is given ordered, persisted
+events from Event Intake. Sometimes when an event is added, it turns out to be the last event that was needed to cause
+an entire "round" of events to come to consensus. When this happens, the Hashgraph module emits a `round`. The round
+includes metadata about the round (the list of witnesses, the round number, the new ancient-round number, etc.) along
```

Comment:
Round don't include a list of witnesses. Rather, they include a list of _**judge** hashes_

They also don't include a new ancient-round number. A round contains its own "round number", and the ancient threshold can be computed from that, given the relevant config value.